crates/volta-core/src/session.rs

```diff
@@ -158,12 +160,14 @@ impl Session {
         }
     }
 
-    pub fn exit(self, code: ExitCode) -> ! {
+    pub fn exit(mut self, code: ExitCode) -> ! {
+        self.event_log.add_event_args();
         self.publish_to_event_log();
         code.exit();
     }
 
-    pub fn exit_tool(self, code: i32) -> ! {
+    pub fn exit_tool(mut self, code: i32) -> ! {
+        self.event_log.add_event_args();
```

Comment:
Super minor: We can probably do this as part of `publish_to_event_log` right before calling `self.event_log.publish`. Would need to update that method to take `mut self`, but that would be it, I think.

Reply:
I had tried to add this to `publish_to_event_log` in that way, but `self.hooks()` is immutably borrowing, and the call to `add_event_args` is also trying to mutably borrow, and I don't see a way to reconcile that.

Reply:
Ahh! Took a quick look and I think we can do it with some destructuring, since `publish_to_event_log` takes ownership of `self`:

```rust
    fn publish_to_event_log(self) {
        let Self {
            project,
            hooks,
            mut event_log,
            ..
        } = self;

        let plugin_res = project
            .get()
            .and_then(|p| hooks.get(p))
            .map(|hooks| hooks.events().and_then(|e| e.publish.as_ref()));

        match plugin_res {
            Ok(plugin) => {
                event_log.add_event_args();
                event_log.publish(plugin);
            }
            Err(e) => {
                debug!("Unable to publish event log.\n{}", e);
            }
        }
    }
```
Basically since we know that the `plugin_res` is only holding references to `self.hooks` and `self.project`, we split the components out into separate variables, then the borrow checker can correctly understand that we aren't crossing any references.

Reply:
Oh, great idea! Didn't even think about destructuring things. That totally works, and I updated to use that in fd3c6de8c6a532eefdcf45d11da727fc90d0cd89